README.md
cabag_extbase
=============

TYPO3 extbase library extension by cab services ag


ChangeLog.txt
2017-02-17 Tizian Schmidlin, Xiaoling Peng <st@cabag.ch>,<xp@cabag.ch>
	* version 1.0.1
	* removed JCR support
	* version 1.0.0
	* fixed php without compatibility6 in TYPO3 7.6

2016-01-20 Tizian Schmidlin <st@cabag.ch>
	* version 0.9.8
	* added some features to query parser through alternative implementation
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php
	* added try and catch to postCheck(), //File Link

2016-01-13 Daniel Kersting <kd@cabag.ch>
	* version 0.9.7
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php
	* fixed issue #24853

2015-08-11 Michael Bossart <mb@cabag.ch>
	* version 0.9.6
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php

2015-07-29 Philip Liebel <ph@cabag.ch>
	* version 0.9.5
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php: modified to display additionalAttributes from link string, working on issue #19750

2015-07-07 Surena Golijani <sg@cabag.ch>
	* version 0.9.4
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php: added parameter cutLinktext to display url without protocol as linktext, working on issue #13540

2015-06-15 Philip Liebel <ph@cabag.ch>
	* version 0.9.3
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php
	* fixed redering of tel links
	* added option to render links to access restricted pages
	* fixed issue #15693
	* fixed issue #17966
	* included the following change from parallel version:

2015-05-08 Michael Bossart <mb@cabag.ch>
	** version 0.9.2.5
	** edited TypoLink-Viewhelper, added posibility to add style-attribute

2015-06-08 Surena Golijani <sg@cabag.ch>
	* version 0.9.2
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php: set parameter returnUrl to returnPart working on issue #16379

2015-05-03 Surena Golijani <sg@cabag.ch>
	* version 0.9.1
	* edited Classes/ViewHelpers/TypoLinkViewHelper.php: added parameter returnUrl to add option to return url only

2015-04-23 Tizian Schmidlin <st@cabag.ch>
	* version 0.9.0
	* [Merged manually from parallel version] fixed issue when sending only plaintext messages

2015-04-13 Michael Bossart <mb@cabag.ch>
	* version 0.8.9
	* fixed bugs in TypoLink-Viewhelper, added check if link to startpage, since no url is generated but link is valid

2015-03-27 Michael Bossart <mb@cabag.ch>
	* version 0.8.8
	* fixed bugs in TypoLink-Viewhelper, added possibility to add a replacment HTML-tag when Link-URL can not be generated

2015-03-10 Philip Liebel <ph@cabag.ch>
	* version 0.8.7
	* worked on issue #14623
	* added option to Tx_CabagExtbase_Utility_Mail to send multipart mails with plaintext version

2015-02-10 Michael Bossart <mb@cabag.ch>
	* version 0.8.6
	* fixed bugs in TypoLink-Viewhelper

2015-02-09 Michael Bossart <mb@cabag.ch>
	* version 0.8.5
	* fixed bugs in TypoLink-Viewhelper

2015-02-03 Tizian Schmidlin <st@cabag.ch>
	* version 0.8.4
	* [6.2] edited Classes/Utility/Mail.php, set useSwiftMailer = true in __construct when TYPO3 6.2+

2015-01-30 Nils Blattner <nb@cabag.ch>
	* version 0.8.3
	* [6.2] edited Classes/TwiceValidator.php, added protected $supportedOptions

2015-01-26 Mnauel Bloch <bm@cabag.ch>
	* version 0.8.2
	* edited Classes/BoundNotEmptyValidator.php, added protected $supportedOptions

2015-01-22 Nils Blattner <nb@cabag.ch>
	* version 0.8.1
	* [BUG] fixed an issue that a flash message was generated when a module token did not validate correctly in Classes/Utility/FixFormToken.php

2015-01-15 Michael Bossart <mb@cabag.ch>
	* version 0.8.0
	* added description for new viewhelper in README.txt

2015-01-14 Michael Bossart <mb@cabag.ch>
	* version 0.7.9
	* added Viewhelper for general typolink-function

2014-12-17 Nils Blattner <nb@cabag.ch>
	* version 0.7.8
	* [FEATURE] finished implementing the eID

2014-12-16 Nils Blattner <nb@cabag.ch>
	* version 0.7.7
	* [TASK] removed closing tags
	* [FEATURE] added an eID script, utility and fixed the mail class to redirect users over an eID into the backend if the link is sent via mail

2014-10-28 Nils Blattner <nb@cabag.ch>
	* version 0.7.6
	* added dontRenderTrusted attribute to the form view helper
	* changed it so the special attributes are no longer rendered to the tag

2014-09-11 Michael Bossart <mb@cabag.ch>
	* version 0.7.5
	* added Viewhelper for if with OR condition

2014-09-11 Tizian Schmidlin <st@cabag.ch>
	* version 0.7.4
	* [GENERAL COMPATIBILITY] added @var to currentLocalName

2014-08-07 Nils Blattner <nb@cabag.ch>
	* version 0.7.3
	* [6.2] replaced 'readXMLLLfile' with 'readLLfile' in Classes/Utility/AbstractWizicon.php

2014-08-07 Nils Blattner <nb@cabag.ch>
	* version 0.7.2
	* [6.2] changed calls to $this->arguments->hasArgument() FormViewHelper (works for 4.5 and 6.2 branches)
	* [6.2] added migration code for Tx_Extbase_Object_Manager

2014-08-04 Nils Blattner <nb@cabag.ch>
	* version 0.7.1
	* [6.2] changed visibility of renderChildren to public in FormViewHelper (works for 4.5 and 6.2 branches)

2014-02-04 Michael Bossart <mb@cabag.ch>
	* version 0.7.0
	* added Viewhelper for link with data-height and data-width attributes

2014-01-29 Nils Blattner <nb@cabag.ch>
	* version 0.6.9
	* [BUG] fixed an issue when the logging receives a non-string message

2013-10-08 Nils Blattner <nb@cabag.ch>
	* version 0.6.8
	* related issue #3752
	* [FEATURE] code cleanup
	*   => Rewrote Tx_CabagExtbase_Domain_Model_StaticLanguage->getCurrentLocalName()
	* [BUG] fixed several security concerns
	*   => replaced Resources/Public/JavaScript/swfupload/Flash/swfupload.swf with wordpress version from http://wordpress.org/download/

2013-09-27 Michael Bossart <mb@cabag.ch>
	* version 0.6.7
	* added Viewhelper for images with lazyloading

2013-09-24 Nils Blattner <nb@cabag.ch>
	* version 0.6.6
	* [BUG] changed pass by ref to pass by value in Tx_CabagExtbase_Utility_Mail::assign()

2013-09-23 Nils Blattner <nb@cabag.ch>
	* version 0.6.5
	* [FEATURE] added an option renderArgumentsAsHidden to the FormViewHelper

2013-09-12 Tizian Schmidlin <st@cabag.ch>
	* version 0.6.4

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "cabag_extbase".
 *
 * Auto generated 22-01-2015 16:23
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
	'title' => 'Extbase Library',
	'description' => 'Extbase library extension',
	'category' => 'plugin',
	'author' => 'Nils Blattner',
	'author_email' => 'nb@cabag.ch',
	'author_company' => 'cab AG',
	'state' => 'stable',
	'uploadfolder' => 0,
	'createDirs' => '',
	'clearCacheOnLoad' => 0,
	'version' => '1.0.1',
	'constraints' =>
	array (
		'depends' =>
		array (
			'cms' => '',
			'extbase' => '',
			'fluid' => '',
		),
		'conflicts' =>
		array (
		),
		'suggests' =>
		array (
		),
	),
	'_md5_values_when_last_written' => 'a:81:{s:13:"ChangeLog.txt";s:4:"7c7c";s:16:"ext_autoload.php";s:4:"0e77";s:21:"ext_conf_template.txt";s:4:"5dbe";s:12:"ext_icon.gif";s:4:"e922";s:17:"ext_localconf.php";s:4:"8082";s:14:"ext_tables.php";s:4:"c85e";s:14:"ext_tables.sql";s:4:"e1c0";s:10:"README.txt";s:4:"b1b1";s:37:"Classes/Controller/BaseController.php";s:4:"3af8";s:38:"Classes/Domain/Model/StaticCountry.php";s:4:"4fe4";s:42:"Classes/Domain/Model/StaticCountryZone.php";s:4:"4346";s:39:"Classes/Domain/Model/StaticLanguage.php";s:4:"0469";s:53:"Classes/Domain/Repository/StaticCountryRepository.php";s:4:"4e92";s:57:"Classes/Domain/Repository/StaticCountryZoneRepository.php";s:4:"24ac";s:54:"Classes/Domain/Repository/StaticLanguageRepository.php";s:4:"6729";s:26:"Classes/Object/Manager.php";s:4:"758b";s:36:"Classes/Persistence/QueryFactory.php";s:4:"008e";s:34:"Classes/Persistence/Repository.php";s:4:"2a0d";s:51:"Classes/Persistence/Storage/CabagTypo3DbBackend.php";s:4:"1c1b";s:35:"Classes/Utility/AbstractWizicon.php";s:4:"c377";s:24:"Classes/Utility/File.php";s:4:"29a6";s:27:"Classes/Utility/Logging.php";s:4:"0557";s:24:"Classes/Utility/Mail.php";s:4:"736b";s:44:"Classes/Validator/BoundNotEmptyValidator.php";s:4:"9238";s:36:"Classes/Validator/TwiceValidator.php";s:4:"3740";s:39:"Classes/ViewHelpers/CdataViewHelper.php";s:4:"1173";s:42:"Classes/ViewHelpers/Date2CalViewHelper.php";s:4:"a242";s:44:"Classes/ViewHelpers/DatepickerViewHelper.php";s:4:"f602";s:39:"Classes/ViewHelpers/DebugViewHelper.php";s:4:"b43b";s:40:"Classes/ViewHelpers/EscapeViewHelper.php";s:4:"3848";s:38:"Classes/ViewHelpers/FormViewHelper.php";s:4:"79c7";s:44:"Classes/ViewHelpers/IfContainsViewHelper.php";s:4:"b9f8";s:41:"Classes/ViewHelpers/IfErrorViewHelper.php";s:4:"9f26";s:38:"Classes/ViewHelpers/IfOrViewHelper.php";s:4:"21cf";s:43:"Classes/ViewHelpers/ImageLazyViewHelper.php";s:4:"625e";s:41:"Classes/ViewHelpers/ImplodeViewHelper.php";s:4:"5829";s:38:"Classes/ViewHelpers/JsonViewHelper.php";s:4:"19a9";s:48:"Classes/ViewHelpers/ObjectAccessorViewHelper.php";s:4:"2959";s:45:"Classes/ViewHelpers/PregReplaceViewHelper.php";s:4:"b9bb";s:37:"Classes/ViewHelpers/RawViewHelper.php";s:4:"22ed";s:40:"Classes/ViewHelpers/SelectViewHelper.php";s:4:"a23c";s:42:"Classes/ViewHelpers/StrftimeViewHelper.php";s:4:"57c1";s:45:"Classes/ViewHelpers/Format/HtmlViewHelper.php";s:4:"b0b7";s:47:"Classes/ViewHelpers/Link/LightboxViewHelper.php";s:4:"092e";s:52:"Classes/ViewHelpers/Widget/FlashuploadViewHelper.php";s:4:"d432";s:63:"Classes/ViewHelpers/Widget/Controller/FlashuploadController.php";s:4:"6337";s:27:"Classes/eID/Flashupload.php";s:4:"f0ca";s:22:"Classes/eID/upload.log";s:4:"5023";s:25:"Configuration/TCA/tca.php";s:4:"eb44";s:34:"Configuration/TypoScript/setup.txt";s:4:"0918";s:33:"Migrations/Code/ClassAliasMap.php";s:4:"87e0";s:40:"Resources/Private/Language/locallang.xml";s:4:"368a";s:43:"Resources/Private/Language/locallang_db.xml";s:4:"ade7";s:44:"Resources/Private/Mail/Logging/Template.html";s:4:"5d37";s:43:"Resources/Private/Mail/Logging/Template.txt";s:4:"9183";s:69:"Resources/Private/Templates/ViewHelpers/Widget/Flashupload/Index.html";s:4:"aea4";s:35:"Resources/Public/Icons/relation.gif";s:4:"e615";s:34:"Resources/Public/Images/button.png";s:4:"9f46";s:52:"Resources/Public/Images/defaultStaticCountryFlag.gif";s:4:"d3e9";s:43:"Resources/Public/JavaScript/fileprogress.js";s:4:"ac2c";s:39:"Resources/Public/JavaScript/handlers.js";s:4:"c0c5";s:36:"Resources/Public/JavaScript/pager.js";s:4:"b5ce";s:40:"Resources/Public/JavaScript/swfupload.js";s:4:"469b";s:46:"Resources/Public/JavaScript/swfupload.queue.js";s:4:"9953";s:56:"Resources/Public/JavaScript/swfupload/Core Changelog.txt";s:4:"d484";s:59:"Resources/Public/JavaScript/swfupload/swfupload license.txt";s:4:"f368";s:50:"Resources/Public/JavaScript/swfupload/swfupload.js";s:4:"3ed0";s:62:"Resources/Public/JavaScript/swfupload/Documentation/index.html";s:4:"057c";s:54:"Resources/Public/JavaScript/swfupload/Flash/deploy.bat";s:4:"1d7a";s:59:"Resources/Public/JavaScript/swfupload/Flash/ExternalCall.as";s:4:"b134";s:55:"Resources/Public/JavaScript/swfupload/Flash/FileItem.as";s:4:"0c76";s:64:"Resources/Public/JavaScript/swfupload/Flash/SWFUpload v2.as3proj";s:4:"f95d";s:56:"Resources/Public/JavaScript/swfupload/Flash/SWFUpload.as";s:4:"12c6";s:57:"Resources/Public/JavaScript/swfupload/Flash/swfupload.swf";s:4:"bd5a";s:70:"Resources/Public/JavaScript/swfupload/Flash/obj/SWFUpload-v2Config.old";s:4:"e86c";s:70:"Resources/Public/JavaScript/swfupload/Flash/obj/SWFUpload-v2Config.xml";s:4:"e86c";s:67:"Resources/Public/JavaScript/swfupload/plugins/SWFObject License.txt";s:4:"5645";s:66:"Resources/Public/JavaScript/swfupload/plugins/swfupload.cookies.js";s:4:"f694";s:64:"Resources/Public/JavaScript/swfupload/plugins/swfupload.queue.js";s:4:"83b6";s:64:"Resources/Public/JavaScript/swfupload/plugins/swfupload.speed.js";s:4:"d840";s:68:"Resources/Public/JavaScript/swfupload/plugins/swfupload.swfobject.js";s:4:"9cf8";}',
);
